egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py

```diff
@@ -67,11 +67,66 @@ def xconfig_line_to_object(config_line, prev_layers = None):
             "*** {0}".format(config_line))
         raise
 
+# This reads raw existing model (*.raw) and returns array of
+# XconfigInputLayer one per input-node or component-node with same 'name' used
+# in raw model and 'dim' equal to 'output-dim' for component-node and input-dim for
+# input-node.
+def get_model_component_info(model_filename):
+    all_layers = []
+    try:
+        f = open(model_filename, 'r')
+    except Exeption as e:
+        sys.exit("{0}: error reading model file '{1}'".format(sys.argv[0], model_filename, repr(e)))
+
+    # use nnet3-info to get component names in the model.
+    out = common_lib.get_command_stdout("""nnet3-info {0} | grep '\-node' """
+                                        """ """.format(model_filename))
+
+    # out contains all component-nodes used in model_filename
+    layer_names = []
+    for line in out.split("\n"):
+        parts = line.split(" ")
+        input_dim = -1
+        output_dim = -1
+        for  field in parts:
+            key_value = field.split("=")
+            if len(key_value) == 2:
+                key = key_value[0]
+                value = key_value[1]
+                if key == "name":
+                    layer_name = value
+                    #layer_name, auxiliary_output = xutils.split_layer_name(value)
+                elif key == "input-dim":
+                    input_dim = int(value)
+                elif key == "output-dim":
+                    output_dim = int(value)
+                elif key == "input":
+                    input_str = value
+
+        if layer_name is not None and layer_name not in layer_names:
+            key_to_value = dict()
+            layer_names.append(layer_name)
+            key_to_value['name'] = layer_name
+            if  input_dim != -1:
+                if output_dim == -1:
+                    # The layer is input layer type.
```

Comment:
I am not sure what this is for. There may never be only input_dim defined with output_dim not defined.